### ISSUE TITLE
Listener on header sticked

### ISSUE THREAD

**Issue by royvangeel** [letsar/flutter_sticky_header#70]
Is it possible to detect when a header is sticked (like using a listener of provide a callback)? I need to create the effect like in the video; I have a header for each weekday, the calendar above the listview needs to be updated based on the sticked header.
I haven't figured out how to do. Thanks!

https://user-images.githubusercontent.com/39674633/143587747-f63970e7-2e1f-4e18-acc2-fd0180ad7e22.mov

**Comment by shw2ypro** [comment]
Hello
Did you find a way to do that? @royvangeel

**Comment by royvangeel** [comment]
@shw2ypro yes, but not using this package. I managed using the sticky_and_expandable_list package and wrap the upper calendar with a StreamBuilder so the ListView pushes the sticked Date into the sink. But I would like to know how to achieve it with this package.

**Comment by letsar** [comment]
Interesting use case

**Comment by daveshirman** [comment]
For anyone needing it, I achieved this by using a `ValueNotifier`, a `ValueListenableBuilder` and `SliverStickyHeader.builder`

**Example: Something like here, where in this example I'm changing a header title that looke like "DECEMBER 2022":**

1. Declare your VLB at the class level, e.g.

`ValueNotifier<String> _currentTitleNotifier = ValueNotifier<String>("---");`

2. Wire up your UI to pump out the _currentTitleNotifier value:

```
ValueListenableBuilder(
    valueListenable: _currentTitleNotifier,
    builder: (context, value, child) {
        return Text(
            value.toString().toUpperCase(),
        );
    },
)
```

3. Observe the state.isPinned in the `SliverStickyHeader.builder(`:

```
builder: (context, state) {
    String newCurrentTitle = DateFormat("MMMM yyyy")
        .format(group.date)
        .toUpperCase();
    if (state.isPinned &&
        _currentTitleNotifier.value.toString() !=
            newCurrentTitle) {
        WidgetsBinding.instance.addPostFrameCallback((_) {
            _currentTitleNotifier.value =
            newCurrentTitle;
        });
    }
}
```

**Comment by IMohamedBassam** [comment]
@daveshirman 
perfect solution.
thanks!
